packages/datadog-plugin-fastify/src/index.js

```diff
@@ -10,9 +13,18 @@ class FastifyPlugin extends RouterPlugin {
   constructor (...args) {
     super(...args)
 
-    this.addSub('apm:fastify:request:handle', ({ req }) => {
+    this.addSub('apm:fastify:request:handle', ({ req, routeConfig }) => {
       this.setFramework(req, 'fastify', this.config)
+      const tags = routeConfig?.[kCodeOriginForSpansTagsSym]
+      if (tags) this.setSpanTags(req, tags)
```

Comment:
Could this be done in a separate plugin instead? The reasoning is that adding product-specific code directly to plugin increases size and complexity even when those products are not used, whereas we could disable product-specific plugins entirely and not even import them when a product is disabled for example.

Reply:
That would actually also remove the need to modify `web` since you could then have something like `setOrigin` on your di-specific base class.

Reply:
> Could this be done in a separate plugin instead?

Do you mean a generic "Code Origin for Spans" plugin that would be used for all the different frameworks? The implementation details for each framework would be slightly different as the way we hook into the router is slightly different. But I guess I can just add a lot of `this.addSub()` calls for the different frameworks we end up supporting in that one generic plugin 🤔

Reply:
> That would actually also remove the need to modify `web` since you could then have something like `setOrigin` on your di-specific base class.

I can of course move the logic for getting the current span based on the current `req` object to a file just related to the plugin, but I thought the whole reason for having the `web` file was to concentrate all the logic in one central place?

Say for example I do create a custom plugin called `datadog-plugin-code-origin`, I could do something like the following code snippet - is that what you had in mind?

```js
const web = require('../../dd-trace/src/plugins/util/web')

// ...

this.addSub('apm:fastify:request:handle', ({ req, routeConfig }) => {
  const tags = routeConfig?.[kCodeOriginForSpansTagsSym]
  if (!tags) return
  const context = web.getContext(req)
  context.span?.addTags(tags)
})
```

Reply:
After trying to implement this with a custom plugin I discovered that the plugin system doesn't seem well suited for plugins that are not actually related to modules being instrumented.

I've just pushed a few commits addressing your comments and I've made a "hack" the current plugin system to make it work. But in the long rung, using plugins for stuff like this, seem to require us to refactor the whole plugin system if we want to do it right - unless of course that I've misunderstood something.

Let's have Zoom call and go over this in person when you have time.

Reply:
Also, just realized that my approach of detecting if the new plugin should be enabled or not doesn't work if you try to enable it programmatically in the call to `tracer.init`  🙄(hence the failing tests now)

Reply:
There's something to be said for having the Fastify related code in `datadog-plugin-fastify` etc. That way it's only executed if Fastify is actually used in the application being instrumented.

That's a point against the idea of having a central `datadog-plugin-code-origin` plugin

Reply:
Based on our Zoom call yesterday, I've now pushed yet another commit that uses the composite plugin approach we discussed. Please let me know what you think.